### PR TITLE
[Transforms] fix Not correctly emitting local name for exported class

### LINKED ISSUE #7864: [Transforms] Not correctly emitting local name for exported class
**Expected**:

```js
var Decl = (function () {
    function Decl() {
    }
    return Decl;
}());
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = Decl;
```

**Actual**:

```js
var Decl = (function () {
    function Decl() {
    }
    return exports.Decl;
}());
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = exports.Decl;
```

Failed tests:
* tests/cases/conformance/es6/modules/defaultExportsCannotMerge02.ts
* tests\cases\compiler\declFileExportImportChain2.ts
* tests/cases/compiler/exportImport.ts

### PR DESCRIPTION
**Pull request by yuit**
Fix #7864

@rbuckton, let me know if this is the correct way to go

### CODE REVIEW

File: src/compiler/transformers/module/module.ts [hunk 1]
```diff
@@ -357,6 +357,9 @@ namespace ts {
                 return undefined;
             }
 
+            // Set emitFlags on the name of the classDeclaration
+            // This is so that when printer will not substitute the identifier
```
Comment (sandersn) [inline src/compiler/transformers/module/module.ts]: 1. Should it be "Set emitFlags on the name of the importEqualsDeclaration" ?
2. I think the second line should be reworded as "This is so the printer ..." ('that when' -> 'the')

Comment (sandersn) [inline src/compiler/transformers/module/module.ts]: Same wording comments on the other two comment changes too

**Comment by sandersn** [comment]
:+1: after minor comment changes

**Comment by mhegazy** [comment]
LGTM

**Pull request merged**